### PR TITLE
New Navigator Interface with Material UI

### PR DESCRIPTION
**Pull request by aravind-segu**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -18,7 +20,7 @@
     "react-landing-page": "^2.0.2",
     "react-redux": "^7.1.3",
     "redux": "^4.0.5",
-    "rich-markdown-editor": "https://github.com/fydp-saac/rich-markdown-editor.git",
+    "rich-markdown-editor": "github:fydp-saac/rich-markdown-editor#master",
```
Comment (Shehan29) [inline package.json:23]: does this allow the editor plugin to be pulled from git rather than local?

File: src/components/SourceEditorWithTagFiltersInput.js [hunk 2]
```diff
@@ -124,7 +124,7 @@ class SourceEditorWithTagFiltersInput extends React.Component {
           this.setState({ fileIdKeyStr, fileContent: value ?? '' });
           FileStorageSystemClient.doGetSourceSavedTagFilters(this.props.currentOpenFileId.sourceId)
             .then(sourceSavedTagFilters => {
-              if (!sourceSavedTagFilters) { alert('failed to retrieve source saved tag filters'); }
+               if (!sourceSavedTagFilters) { }//alert('failed to retrieve source saved tag filters'); }
```
Comment (Shehan29) [inline src/components/SourceEditorWithTagFiltersInput.js:127]: clean this up?

File: src/backend/HostedFileStorageSystemClient.js [hunk 3]
```diff
@@ -21,7 +21,7 @@ export default {
     const token = Cookies.get(ACCESS_TOKEN_COOKIE_KEY);
     try {
       const response = await fetchWithTimeout(
-        SERVER_BASE_URL + `view?docID=${sourceId}` + (!createNew ? `&viewID=${viewId}&type=${type}` : ''),
+        SERVER_BASE_URL + `view?docID=${sourceId}&viewType=${type}` + (!createNew ? `&viewID=${viewId}` : ''),
```
Comment (AkshayPall) [inline src/backend/HostedFileStorageSystemClient.js:24]: Ah, I guess these changes relate to your other PR

File: src/components/Navigator.js [hunk 4]
```diff
@@ -197,7 +223,7 @@ class Navigator extends React.Component {
       alert('failed to create new file');
       return;
     }
-    const updatedSourceId = sourceFileType ? newFile.id : this.props.currentOpenFileId.sourceId;
+    const updatedSourceId = sourceFileType ? newFile.id : this.state.selectedFileId;
```
Comment (Shehan29) [inline src/components/Navigator.js:226]: what's the reason for this change?

File: yarn.lock [hunk 5]
```diff
@@ -1195,7 +1195,7 @@
   dependencies:
     regenerator-runtime "^0.13.4"
 
-"@babel/runtime@^7.12.1", "@babel/runtime@^7.4.2", "@babel/runtime@^7.6.3", "@babel/runtime@^7.8.7":
+"@babel/runtime@^7.12.1", "@babel/runtime@^7.3.1", "@babel/runtime@^7.4.2", "@babel/runtime@^7.4.4", "@babel/runtime@^7.6.3", "@babel/runtime@^7.8.3", "@babel/runtime@^7.8.7":
```
Comment (AkshayPall) [inline yarn.lock:1198]: Why did this regress? I'm guessing this was automatic? weird.

File: yarn.lock [hunk 6]
```diff
@@ -10125,10 +10317,9 @@ rgba-regex@^1.0.0:
   resolved "https://registry.yarnpkg.com/rgba-regex/-/rgba-regex-1.0.0.tgz#43374e2e2ca0968b0ef1523460b7d730ff22eeb3"
   integrity sha1-QzdOLiyglosO8VI0YLfXMP8i7rM=
 
-"rich-markdown-editor@https://github.com/fydp-saac/rich-markdown-editor.git":
+"rich-markdown-editor@github:fydp-saac/rich-markdown-editor#master":
```
Comment (AkshayPall) [inline yarn.lock:10320]: holy shit this works?!?!?

File: yarn.lock [hunk 7]
```diff
@@ -10125,10 +10317,9 @@ rgba-regex@^1.0.0:
...
   version "10.0.1"
-  uid c7b0ff1f85fd76d36a40c58f1c3a1dcf0c2d6c41
-  resolved "https://github.com/fydp-saac/rich-markdown-editor.git#c7b0ff1f85fd76d36a40c58f1c3a1dcf0c2d6c41"
+  resolved "https://codeload.github.com/fydp-saac/rich-markdown-editor/tar.gz/c7b0ff1f85fd76d36a40c58f1c3a1dcf0c2d6c41"
```
Comment (AkshayPall) [inline yarn.lock:10322]: wow

**Review by AkshayPall — APPROVED** [review]

**Review by Shehan29 — APPROVED** [review]
🚢

File: src/backend/HostedFileStorageSystemClient.js (hunk 3 above)
Comment (aravind-segu) [reply src/backend/HostedFileStorageSystemClient.js:24]: Yes, the url doesnt take viewID for createNew

File: yarn.lock (hunk 5 above)
Comment (aravind-segu) [reply yarn.lock:1198]: Its automatic, idk. Tbh should we even push yarn.lock?

File: yarn.lock (hunk 6 above)
Comment (aravind-segu) [reply yarn.lock:10320]: No It doesnt, still the same problem.

File: package.json (hunk 1 above)
Comment (aravind-segu) [reply package.json:23]: nope, for some reason it doesnt work. i also added a post install script and that didnt work too :(

File: src/components/Navigator.js (hunk 4 above)
Comment (aravind-segu) [reply src/components/Navigator.js:226]: idk how this worked earlier. But basically above this code we close the "current file" when opening a new one. Then currentOpenFileId is set to None. But i already have it stored in the state for highlighting purposes, so i just used that.

File: yarn.lock (hunk 5 above)
Comment (AkshayPall) [reply yarn.lock:1198]: not sure, but if it's already included then maybe it's fine?

File: src/components/Navigator.js [hunk 8]
```diff
@@ -90,6 +120,12 @@ const DEFAULT_STATE = {
   renaming: NO_RENAMING_STATE,
   filesList: {},
   nextNewFileIds: null,
+  selectedFileId: null,
+  selectedViewId: null,
+  selectedFileOpen: false,
```
Comment (ca-shen98) [inline src/components/Navigator.js:125]: Why did we add this state - it should already be in the currentOpenFIleId props state. Now there's a possibility of inconsistency. We should be using the redux state since it should be the source of truth for what the state of which file is open.

File: src/components/Navigator.js (hunk 4 above)
Comment (ca-shen98) [reply src/components/Navigator.js:226]: The way it worked earlier was when it closed the open file, the state change was propagated via redux, and then in the componentDidUpdate, it updates the value of the displayed names. I think this state change flow is much more consistent - see my previous comment about not having duplicate sources of state.

File: src/components/Navigator.js [hunk 9]
```diff
@@ -342,14 +364,6 @@ class Navigator extends React.Component {
...
       });
     }
-    if (
-      prevProps.currentOpenFileId.sourceId !== this.props.currentOpenFileId.sourceId ||
-      prevProps.currentOpenFileId.viewId !== this.props.currentOpenFileId.viewId ||
-      prevState.filesList !== this.state.filesList
-    ) {
-      document.getElementById(CURRENT_SOURCE_NAME_INPUT_ID).value = this.getSourceName(this.props.currentOpenFileId);
-      document.getElementById(CURRENT_VIEW_NAME_INPUT_ID).value = this.getViewName(this.props.currentOpenFileId);
-    }
```
Comment (ca-shen98) [inline src/components/Navigator.js:352]: Here is where the displayed names are updated.
